ivy/func_wrapper.py

```diff
@@ -812,6 +812,14 @@ def _handle_device_shifting(*args, **kwargs):
         -------
             The return of the function.
         """
+        if "device" in kwargs and kwargs["device"] is not None:
+            out = fn(*args, **kwargs)
```

Comment:
let's rename this to `ret` to avoid confusion with the `out` argument
Secondly, I think this implementation would simply bypass the device shifting logic when we have a `device` argument in `kwargs`, our handling would be different based on whether soft device was enabled or not. 
I'll request a review from @sherry30, thanks @abdulasiraj 😄